src/components/contact/components/contact-form/ContactForm.tsx

```diff
@@ -0,0 +1,131 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { useMutation } from '@tanstack/react-query';
+import { type ContactSchema, contactSchema } from '@/schemas/contact.schema';
+
+export const ContactForm: React.FC = () => {
+  const {
+    handleSubmit,
+    register,
+    setError,
+    formState: { errors },
+    reset,
+  } = useForm<ContactSchema>({
+    resolver: zodResolver(contactSchema),
+  });
+
+  const submitFormData = async (
+    data: ContactSchema,
+  ): Promise<{ message: string }> => {
+    const response = await fetch('/api/contact', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (!response.ok) {
+      const errorData = await response.json();
+
+      throw new Error(JSON.stringify(errorData));
+    }
+
+    return response.json();
+  };
+
+  const { mutate } = useMutation({
+    mutationFn: submitFormData,
+    onSuccess: () => {
+      alert('Form submitted successfully');
+      reset();
+    },
+    onError: (error) => {
+      try {
+        const errorData = JSON.parse(error.message);
+
+        Object.keys(errorData.errors).forEach((key) => {
+          setError(key as keyof ContactSchema, {
+            message: errorData[key][0],
+          });
+        });
+      } catch (error) {
+        alert('An error occurred while submitting the form');
+      }
+    },
+  });
```

Comment:
mutation should be defined in src/contact/queries.ts

Reply:
moved the mutation to the queries.ts